components/SwiperProgressBar.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <div class="progress-container">
+    <div v-for="i in pages" :key="i" ref="progress" class="progress"></div>
```

Comment:
```suggestion
    <div v-for="i in pages" :key="`progress-key-${i}" ref="progress" class="progress"></div>
```

Reduce the chance of conflicting keys